pilot/pkg/security/authn/utils/utils.go

```diff
@@ -37,7 +37,7 @@ const (
 
 // BuildInboundFilterChain returns the filter chain(s) corresponding to the mTLS mode.
 func BuildInboundFilterChain(mTLSMode model.MutualTLSMode, sdsUdsPath string, node *model.Proxy,
-	listenerProtocol networking.ListenerProtocol) []networking.FilterChain {
+	listenerProtocol networking.ListenerProtocol, push *model.PushContext) []networking.FilterChain {
```

Comment:
I was thinking to only pass the list of trust domains (union of the local one and the aliases).

This will be simliar to other parameter, more specific and minimal necessary.

Reply:
Yeah, I've thought about it too- shouldn't be an issue to change this up. But maybe it's a better idea to pass in pushcontext for this to be consistent with Gateway functions? Also maybe it's more "future-proof" as we could have more meshconfig options that we might need. @howardjohn what implementation do you prefer